diff --git a/compileopts/config.go b/compileopts/config.go
index 9349738a..e3f67b11 100644
--- a/compileopts/config.go
+++ b/compileopts/config.go
@@ -310,7 +310,11 @@ func (c *Config) CFlags(libclang bool) []string {
 		)
 	case "wasi-libc":
 		root := goenv.Get("TINYGOROOT")
-		cflags = append(cflags, "--sysroot="+root+"/lib/wasi-libc/sysroot")
+		cflags = append(cflags,
+			"-nostdlibinc",
+			"-isystem", root+"/lib/wasi-libc/sysroot/include")
+	case "wasmbuiltins":
+		// nothing to add (library is purely for builtins)
 	case "mingw-w64":
 		root := goenv.Get("TINYGOROOT")
 		path, _ := c.LibcPath("mingw-w64")
